dds/idl/typeobject_generator.cpp

```diff
@@ -11,339 +11,1593 @@
+  be_global->impl_ << ",";
+  print(*array_sdefn.element_identifier);
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::PlainArrayLElemDefn& array_ldefn)
+{
+  be_global->impl_ << "XTypes::PlainArrayLElemDefn(";
+  print(array_ldefn.header);
+  be_global->impl_ << ",";
+  print(array_ldefn.array_bound_seq);
+  be_global->impl_ << ",";
+  print(*array_ldefn.element_identifier);
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::PlainMapSTypeDefn& map_sdefn)
+{
+  be_global->impl_ << "XTypes::PlainMapSTypeDefn(";
+  print(map_sdefn.header);
+  be_global->impl_ << ",";
+  print_sbound(map_sdefn.bound);
+  be_global->impl_ << ",";
+  print(*map_sdefn.element_identifier);
+  be_global->impl_ << ",";
+  print_collection_element_flags(map_sdefn.key_flags);
+  be_global->impl_ << ",";
+  print(*map_sdefn.key_identifier);
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::PlainMapLTypeDefn& map_ldefn)
+{
+  be_global->impl_ << "XTypes::PlainMapLTypeDefn(";
+  print(map_ldefn.header);
+  be_global->impl_ << ",";
+  print_lbound(map_ldefn.bound);
+  be_global->impl_ << ",";
+  print(*map_ldefn.element_identifier);
+  be_global->impl_ << ",";
+  print_collection_element_flags(map_ldefn.key_flags);
+  be_global->impl_ << ",";
+  print(*map_ldefn.key_identifier);
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::EquivalenceHash& equivalence_hash)
+{
+  be_global->impl_
+    << '{'
+    << static_cast<int>(equivalence_hash[0])
+    << ','
+    << static_cast<int>(equivalence_hash[1])
+    << ','
+    << static_cast<int>(equivalence_hash[2])
+    << ','
+    << static_cast<int>(equivalence_hash[3])
+    << ','
+    << static_cast<int>(equivalence_hash[4])
+    << ','
+    << static_cast<int>(equivalence_hash[5])
+    << ','
+    << static_cast<int>(equivalence_hash[6])
+    << ','
+    << static_cast<int>(equivalence_hash[7])
+    << ','
+    << static_cast<int>(equivalence_hash[8])
+    << ','
+    << static_cast<int>(equivalence_hash[9])
+    << ','
+    << static_cast<int>(equivalence_hash[10])
+    << ','
+    << static_cast<int>(equivalence_hash[11])
+    << ','
+    << static_cast<int>(equivalence_hash[12])
+    << ','
+    << static_cast<int>(equivalence_hash[13])
+    << '}';
+}
+
+void
+print(const OpenDDS::XTypes::TypeObjectHashId& id)
+{
+  be_global->impl_ << "XTypes::TypeObjectHashId(";
+  print_equivalence_kind(id.kind);
+  be_global->impl_ << ",";
+  print(id.hash);
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::StronglyConnectedComponentId& scc)
+{
+  be_global->impl_ << "XTypes::StronglyConnectedComponentId(";
+  print(scc.sc_component_id);
+  be_global->impl_ << ", " << scc.scc_length << ", " << scc.scc_index;
+  be_global->impl_ << ")";
+}
+
+void
+print(const OpenDDS::XTypes::TypeIdentifier& ti)
+{
+  switch (ti.kind()) {
+  case OpenDDS::XTypes::TK_NONE:
+    be_global->impl_ << "XTypes::TypeIdentifier(XTypes::TK_NONE)";
```

Comment:
Move the common prefix to before the `switch`?  Could avoid repetition in each branch.

Also should we have a function in TypeObject.h that gives you the stringized `TK_*`?

Reply:
Yes on the switch.

It should be easy to pull out when we need such a function.